packages/itwin/map-layers/src/ui/FeatureInfoUiItemsProvider.tsx

```diff
@@ -32,6 +34,26 @@ export const getMapFeatureInfoToolItemDef = (): ToolItemDef =>
       await IModelApp.tools.run(MapFeatureInfoTool.toolId);
     },
     badgeType: BadgeType.TechnicalPreview,
+    isHidden: new ConditionalBooleanValue(() => {
+      // Hide the MapFeatureInfoTool if the Map Layers toggle is off or no ArcGISFeature layers are active
+      const vp = IModelApp.viewManager.selectedView;
+      if (!vp || !vp.viewFlags.backgroundMap) {
+        return true;
+      }
+      const backgroundLayers = vp.displayStyle.settings.mapImagery.backgroundLayers.map((value) => value.toJSON());
+      for (const layer of backgroundLayers) {
+        if (layer.formatId === "ArcGISFeature") {
+          return false;
+        }
+      }
+      const overlayLayers = vp.displayStyle.settings.mapImagery.overlayLayers.map((value) => value.toJSON());
+      for (const layer of overlayLayers) {
+        if (layer.formatId === "ArcGISFeature") {
+          return false;
+        }
+      }
+      return true;
+    }, [SyncUiEventId.ViewStateChanged]),
```

Comment:
this event is very chatty, it is triggered from the viewport's onViewChanged
```typescript
/** Event called whenever this viewport is synchronized with its [[ViewState]].
     * @note This event is invoked *very* frequently. To avoid negatively impacting performance, consider using one of the more specific Viewport events;
     * otherwise, avoid performing excessive computations in response to this event.
     * @see [[onViewportChanged]] for receiving events at more regular intervals with more specific information about what changed.
     * @see [[onChangeView]] for an event raised specifically when a different [[ViewState]] becomes associated with the viewport.
     */
    readonly onViewChanged: BeEvent<(vp: Viewport) => void>;
```
Probably not the best thing to do given the `toJSON()` operation here, it could contribute to poor performance, 
we really would want a syncuievent associated with the `onDisplayStyleChanged` but I'm not seeing one that exists yet. Perhaps though a unique syncuievent for map layers when a layer is attached in conjunction with the `SyncUiEventId.ActiveViewportChanged` would cover this case for the isHidden to work as expected

Reply:
Switched from using SyncUiEventId.ViewStateChanged to custom event MapFeatureInfoTool.toolSyncUiEventId to trigger isHidden for MapFeatureInfoTool

Reply:
Wanted to write out what we discussed in a call earlier so @mdastous-bentley can weigh in when he gets back

- regarding the `MapFeatureInfoTool.toolSyncUiEventId`, we want the `SyncUiEvent` we use here to be more generalized to the event we care about, which technically is the [onMapImageryChanged](https://github.com/iTwin/itwinjs-core/blob/master/core/common/src/DisplayStyleSettings.ts#L473). Unfortunately, there isn't a `SyncUiEvent` for this yet. It probably should live next to other view related events in appui's `SyncUiEventDispatcher` you can find [here](https://github.com/iTwin/appui/blob/release/4.16.x/ui/appui-react/src/appui-react/syncui/SyncUiEventDispatcher.ts#L25), but perhaps since this package is the general 'owner' of these settings, it wouldn't be completely inappropriate for this package to export its own `MapLayersSyncUiEventId` enum with `MapImageryChanged` that it dispatches itself from a useEffect in [MapLayersWidget](https://github.com/iTwin/viewer-components-react/blob/3cf4d181df078b62ea0484624fe2737f22243392/packages/itwin/map-layers/src/ui/widget/MapLayersWidget.tsx). Then the tool here could use that `SyncUiEvent` in addition to `SyncUiEventId.ActiveViewportChanged` to recheck whether to hide this tool
- regarding the `MapFeatureInfoTool`, I'm unsure how appropriate hardcoding `"ArcGIS"` here is. Isn't this tool generalized and not intended to be coupled to specific formatIds? If so would it be better to check against the registered formatId's? Is there an additional attribute of any registered format that makes them relevant to this tool?

Reply:
This is a side note, but I was looking into this and `MapLayerImageryProvider.supportsMapFeatureInfo` is alpha right now. Considering the recent effort to no longer use internal/alpha APIs outside of the core repo, I would suggest we promote it to beta first or alongside this PR. This seems like a good use case of having access to that property

Reply:
I need to get supportsMapFeatureInfo upgraded to beta to clear this error:

D:\a\1\s\packages\itwin\map-layers\src\ui\FeatureInfoUiItemsProvider.tsx(47,13): error @itwin/no-internal : getter "MapLayerImageryProvider.supportsMapFeatureInfo" is alpha.